### ISSUE TITLE
OOB Handling of both 1.0 and 1.1 messages

### ISSUE THREAD

**Issue by swcurran** [hyperledger/aries-cloudagent-python#1917]
It appears that ACA-Py is not handling OOB messages for both protocol version 1.0 and 1.1, as it should be.  Per [Aries RFC 0003](https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0003-protocols#semver-rules-for-protocols), ACA-Py should be able to handle all minor versions of a protocol.  Please investigate.

See [this comment on an AFJ](https://github.com/hyperledger/aries-framework-javascript/pull/930#issuecomment-1228897006) issue with details about what is happening.

@JamesKEbert -- could you please add steps to reproduce?

Thanks!

**Comment by swcurran** [comment]
@shaangill025 -- can you please take a look at this when you have some time?

First off, it would be nice to know what version(s) of the OOB protocol that ACA-Py supports, and where is the error happening?

Thanks!

**Comment by JamesKEbert** [comment]
The steps to reproduce are a little less straightforward unfortunately, as this was created using a combination of the Aries Toolbox + AFJ running in a RN context. 

The biggest thing to note is we utilized the ACA-Py configs `public-invites: true` and `seed: PLACE_SEED_HERE` and then posted the agent endpoint with the API endpoint `wallet/set-did-endpoint`.

We created two OOB invitations via `/out-of-band/create-invitation` with the params `auto_accept:true` and `multi_use:false` and the following body:
```
{
  "alias": "TestAgent",
  "attachments": [
  ],
  "handshake_protocols": [
    "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0"
  ],
  "my_label": "Toolbox-Public-DID",
  "use_public_did": true
}
```

The first invitation is used to establish the connection, and the second invitation to trigger a connection reuse.

We could get a Mobile Agent APK created either today or Monday with the mobile behavior from the [AFJ 930 PR](https://github.com/hyperledger/aries-framework-javascript/pull/930). 

Although now that I think about it, @swcurran is the connection reuse case handled in AATH?
